File: Algorithms/Search/BinarySearch/BinarySearch.java

class BinarySearch {
    //Iterative version
    int search(int arr[], int x){
        int left = 0, mid;
        int right = arr.length - 1;

        while(right >= left){
            mid = (left + right) / 2;

            if(arr[mid] == x)
                return mid;
            else if(arr[mid] > x)
                right = mid - 1;
            else
                left = mid + 1;
        }
        return -1;
    }
  
} 

File: Algorithms/Search/BinarySearch/BinarySearch.py

#Recursive version
def binarySearch(arr,x,left,right):
    if(right > 0):
        #find mid of arr
        mid = left + (right - left) // 2
        if(arr[mid] == x):
            return mid;
        elif(x > arr[mid]):
            return binarySearch(arr, x, mid + 1, right)
        else:
            #x < arr[mid]
            return binarySearch(arr, x, left, mid - 1)
    return -1
